pkg/kube/pod.go

```diff
@@ -36,35 +38,51 @@ type PodOptions struct {
 	Command            []string
 	Volumes            map[string]string
 	ServiceAccountName string
+	PodOverride        v1.PodSpec
 }
 
 // CreatePod creates a pod with a single container based on the specified image
-func CreatePod(ctx context.Context, cli kubernetes.Interface, opts *PodOptions) (*v1.Pod, error) {
+func CreatePod(ctx context.Context, cli kubernetes.Interface, opts *PodOptions) (pod *v1.Pod, err error) {
 	volumeMounts, podVolumes := createVolumeSpecs(opts.Volumes)
-	pod := &v1.Pod{
+	defaultSpecs := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:            "container",
+				Image:           opts.Image,
+				Command:         opts.Command,
+				ImagePullPolicy: v1.PullPolicy(v1.PullAlways),
+				VolumeMounts:    volumeMounts,
+			},
+		},
+		// RestartPolicy dictates when the containers of the pod should be restarted.
+		// The possible values include Always, OnFailure and Never with Always being the default.
+		// OnFailure policy will result in failed containers being restarted with an exponential back-off delay.
+		RestartPolicy:      v1.RestartPolicyOnFailure,
+		Volumes:            podVolumes,
+		ServiceAccountName: opts.ServiceAccountName,
+	}
+	// Override default specs if podspecs are passed
+	if !reflect.DeepEqual(opts.PodOverride, v1.PodSpec{}) {
```

Comment:
rather than repeat this check in each function, can we push this logic down into `PodSpecOverride`?